presto-main/src/main/java/io/prestosql/execution/buffer/PagesSerde.java

```diff
@@ -43,52 +45,73 @@
     private final BlockEncodingSerde blockEncodingSerde;
     private final Optional<Compressor> compressor;
     private final Optional<Decompressor> decompressor;
+    private final Optional<SpillCipher> spillCipher;
 
-    public PagesSerde(BlockEncodingSerde blockEncodingSerde, Optional<Compressor> compressor, Optional<Decompressor> decompressor)
+    public PagesSerde(BlockEncodingSerde blockEncodingSerde, Optional<Compressor> compressor, Optional<Decompressor> decompressor, Optional<SpillCipher> spillCipher)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
+        checkArgument(compressor.isPresent() == decompressor.isPresent(), "compressor and decompressor must both be present or both be absent");
         this.compressor = requireNonNull(compressor, "compressor is null");
         this.decompressor = requireNonNull(decompressor, "decompressor is null");
-        checkArgument(compressor.isPresent() == decompressor.isPresent(), "compressor and decompressor must both be present or both be absent");
+        this.spillCipher = requireNonNull(spillCipher, "spillCipher is null");
```

Comment:
PagesSerde can be used for many things other than spilling , so "spill cipher" doesn't seem like the be the right name (for the field and for the interface name).
Maybe call the interface `PageCipher` (and the field, impl accordingly)?

Reply:
As discussed offline, renaming this to `PageCipher` implies that it's usable in a very broad context, whereas `SpillCipher` is more specific to it's function and indeed- limitations. These are ephemeral one-use ciphers that can only be used for reading and writing from disk from within the same process. I'll be leaving it `SpillCipher` for now pending any further arguments for / against another name.